src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -312,12 +315,70 @@ struct CPUStreamsExecutor::Impl {
         std::vector<int> _cpu_ids;
 #endif
     };
+    // if the thread is created by CPUStreamsExecutor, the Impl::Stream of the thread is stored by tbb Class
+    // enumerable_thread_specific, the alias is ThreadLocal, the limit of ThreadLocal please refer to
+    // https://spec.oneapi.io/versions/latest/elements/oneTBB/source/thread_local_storage/enumerable_thread_specific_cls.html
+    // if the thread is created by customer, the Impl::Stream of the thread will be stored in variable _stream_map, and
+    // will be count by thread_local t_stream_count_map when the customer's thread is destoried, the stream's count will
+    // became 1 and local() will reuse one of them, and release others.
+    class CustomThreadLocal : public ThreadLocal<std::shared_ptr<Stream>> {
```

Comment:
Can we add new tests?

Reply:
@ilyachur  I  try to add test case in [https://github.com/openvinotoolkit/openvino/blob/master/tests/stress_tests/memleaks_tests/test.cpp](https://github.com/openvinotoolkit/openvino/blob/master/tests/stress_tests/memleaks_tests/tests.cpp), but the leak is very small, it's hard to set good threshold.

Reply:
@ilyachur I requested memleak tests on Win in precommit in #111491, it should be enough to cover such case

Reply:
@tiger100256-hu please, find updated test in https://github.com/vurusovs/openvino/tree/vu/win_memleak_test. It detects issue on master, and pass with your changes

Reply:
@vurusovs  thanks very much, I think your test case is better, I didn't find a good threshold. Maybe this PR only fix the issue without test case and you can submit test case pr in ticket #111491.  @ilyachur do you agree that?

Reply:
let's merge test in separate PR because it requires review of another team (so it's very time consuming)

Reply:
@ilyachur @wangleis  help to go on review, thanks